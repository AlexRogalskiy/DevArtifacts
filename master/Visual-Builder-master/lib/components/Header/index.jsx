/**
 * External dependencies
 */
import React, { Component, PropTypes } from 'react';
import classNames from 'classnames';

/**
 * Internal dependencies
 */
import './style.scss';
import Button from '../Button';
import PagesMenu from '../PagesMenu';
import ResponsiveMenu from '../ResponsiveMenu';

/**
 * <Header />
 */
class Header extends Component {
  render() {
    const classes = classNames('header', this.props.className);

    return (
      <header className={classes} role="banner">
        <div className="header__group">
          <PagesMenu
            addPage={this.props.pagesMenu.actions.addPage}
            changePage={this.props.pagesMenu.actions.changePage}
            deletePage={this.props.pagesMenu.actions.deletePage}
            pages={this.props.pagesMenu.state.pages}
            toggleMenu={this.props.pagesMenu.actions.toggleMenu}
            status={this.props.pagesMenu.state.status}
          />
          <ResponsiveMenu
            changeContentSize={this.props.responsiveMenu.actions.changeContentSize}
            contentSize={this.props.responsiveMenu.state.contentSize}
            decreaseContentSize={this.props.responsiveMenu.actions.decreaseContentSize}
            increaseContentSize={this.props.responsiveMenu.actions.increaseContentSize}
            toggleMenu={this.props.responsiveMenu.actions.toggleMenu}
            status={this.props.responsiveMenu.state.status}
          />
        </div>
        <div className="header__group">
          <Button>
            <span className="ion ion-ios-cloud-download" />
          </Button>
          <Button>
            <span className="ion ion-gear-a" />
          </Button>
        </div>
      </header>
    );
  }
}

export default Header;
