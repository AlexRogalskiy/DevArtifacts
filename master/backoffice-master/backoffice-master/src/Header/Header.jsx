import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'
import {
  AppBar,
  IconButton,
  Toolbar,
  Typography,
  withStyles,
} from '@material-ui/core'
import MenuIcon from '@material-ui/icons/Menu'

const drawerWidth = 280

const styles = theme => ({
  root: {
    width: '100%',
  },
  appBar: {
    transition: theme.transitions.create(['margin', 'width'], {
      easing: theme.transitions.easing.sharp,
      duration: theme.transitions.duration.leavingScreen,
    }),
    transform: 'translate(0, 0)',
  },
  appBarWithCookieInfo: {
    transform: `translate(0, ${theme.spacing.unit * 6}px)`,
  },
  appBarShift: {
    width: `calc(100% - ${drawerWidth}px)`,
    transition: theme.transitions.create(['margin', 'width'], {
      easing: theme.transitions.easing.easeOut,
      duration: theme.transitions.duration.enteringScreen,
    }),
    marginLeft: drawerWidth,
  },
  menuButton: {
    marginLeft: -12,
    marginRight: 20,
  },
  hide: {
    display: 'none',
  },
  flex: {
    flex: 1,
  },
  title: {
    cursor: 'pointer',
    '&:hover': {
      textDecoration: 'underline',
    },
  },
})

const Header = ({
  title,
  isOpen,
  isFixed,
  onDrawerOpen,
  isCookieInfoOpen,
  onClick,
  children,
  classes,
}) => (
  <div className={classes.root}>
    <AppBar
      className={classNames(classes.appBar, {
        [classes.appBarShift]: isOpen,
        [classes.appBarWithCookieInfo]: isCookieInfoOpen,
      })}
      position={isFixed ? 'fixed' : 'static'}
    >
      <Toolbar>
        <IconButton
          aria-label="open drawer"
          color="inherit"
          onClick={onDrawerOpen}
          className={classNames(classes.menuButton, isOpen && classes.hide)}
        >
          <MenuIcon />
        </IconButton>

        <Typography
          variant="title"
          color="inherit"
          className={`${classes.flex} ${classes.title}`}
          onClick={onClick}
        >
          {title}
        </Typography>

        {children}
      </Toolbar>
    </AppBar>
  </div>
)

Header.propTypes = {
  isOpen: PropTypes.bool,
  title: PropTypes.string.isRequired,
  isFixed: PropTypes.bool.isRequired,
  isCookieInfoOpen: PropTypes.bool.isRequired,
  onDrawerOpen: PropTypes.func.isRequired,
  onClick: PropTypes.func.isRequired,
  children: PropTypes.node,
  classes: PropTypes.objectOf(PropTypes.string).isRequired,
}

Header.defaultProps = {
  isOpen: false,
  children: null,
}

export default withStyles(styles)(Header)
