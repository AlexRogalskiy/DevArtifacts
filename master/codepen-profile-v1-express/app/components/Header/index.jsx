/**
 * External dependencies
 */
import React, { PropTypes } from 'react';
import cx from 'classnames';
import { Link } from 'react-router';

/**
 * Internal dependencies
 */
import './style.scss';
import { Container } from '../Grid';
import Logo from '../Logo';
import Nav from '../Nav';

/**
 * Local variables
 */
const propTypes = {
  className: PropTypes.string,
  fixed: PropTypes.bool,
  style: PropTypes.object,
  windowWidth: PropTypes.number,
};

const defaultProps = {
  fixed: false,
};

const Header = (props) => {
  let classes = cx({
    header: true,
    'header--fixed': props.fixed,
  }, props.className);

  let menu = [
    [
      {
        index: true,
        url: '/',
        children: 'Pens',
      },
      {
        url: '/posts',
        children: 'Posts',
      },
      {
        url: '/collections',
        children: 'Collections',
      },
    ],
  ];

  let menuTwo = [
    [
      {
        anchor: true,
        url: 'http://codepen.io/andytran',
        children: 'Follow on CodePen &#8594;',
      },
    ],
  ];

  return (
    <header className={classes}>
      <Container>
        <div className="header__content">
          {props.windowWidth >= 768 || !props.windowWidth ? <Nav menus={menu} /> : false}
          <Link to="/">
            <Logo color="#2C2C2C" />
          </Link>
          <Nav menus={menuTwo} />
        </div>
      </Container>
    </header>
  );
};

Header.propTypes = propTypes;
Header.defaultProps = defaultProps;

export default Header;
