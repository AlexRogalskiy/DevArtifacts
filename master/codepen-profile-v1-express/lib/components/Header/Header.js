/**
 * External dependencies
 */
import React, { PropTypes } from 'react';
import { includes } from 'lodash';

/**
 * Internal dependencies
 */
import { capitalizeFirstLetter } from '../../helpers';
import Button from '../Button/Button';
import ButtonGroup from '../ButtonGroup/ButtonGroup';
import Tabs from '../Tabs/Tabs';

/**
 * Lcoal variables
 */
const propTypes = {
  params: PropTypes.object,
};

const defaultProps = {};

/**
 * <Header />
 */
const Header = ({ params }) => (
  <header className="header">
    <div className="header__top">
      <h1 className="header__title">{params.view ? capitalizeFirstLetter(params.view) : false}</h1>
      <div className="header__content">
        test
      </div>
    </div>
    <div className="header__bottom">
      {includes(['collections', 'pens', 'posts'], params.view) ? <Tabs params={params} /> : false}
    </div>
    <ButtonGroup
      style={{
        position: 'absolute',
        right: '40px',
        top: '136px',
        width: '198px',
      }}
      vertical
    >
      <Button block primary>Button</Button>
      <Button block primary>Button</Button>
    </ButtonGroup>
  </header>
);

Header.propTypes = propTypes;
Header.defaultProps = defaultProps;

export default Header;
