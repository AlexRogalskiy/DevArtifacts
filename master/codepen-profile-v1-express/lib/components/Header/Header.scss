// Header
.header {
  position: relative;
  height: 136px;
  padding: 0 40px;
  box-sizing: border-box;

  &:before {
    content: ' ';
    position: absolute;
    right: 278px;
    bottom: 0;
    left: 40px;
    display: block;
    height: 1px;
    box-shadow: 0 1px #EFEFEF;
  }

  &__top {
    display: flex;
    flex-direction: row;
    padding: 40px 0 0;
  }

  &__title {
    margin: 0;
    padding: 0;
    font-size: 1.25rem;
    font-weight: $font-regular;
    line-height: 2rem;
  }

  &__content {
    flex-grow: 1;
  }

  &__bottom {
    position: absolute;
    bottom: 0;
  }
}
