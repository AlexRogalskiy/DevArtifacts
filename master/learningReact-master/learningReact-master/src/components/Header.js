import React from 'react';

const Header = (props) => {
	return (
		<header className="top">
			<h1>Catch
			<span className="ofThe">
				<span className="of">of</span>
				<span className="the">the</span>
			</span>
			Day</h1>
			<h3 className="tagline">{props.tagline}</h3>
		</header>
	)
}

Header.propTypes = {
	tagline: React.PropTypes.string.isRequired
};

export default Header;
