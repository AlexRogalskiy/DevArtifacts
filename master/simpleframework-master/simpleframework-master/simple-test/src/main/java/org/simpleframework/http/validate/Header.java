package org.simpleframework.http.validate;

import org.simpleframework.xml.util.Entry;

public interface Header extends Entry{        
   public String getValue();  
   public boolean nameMatches(String str);
}

