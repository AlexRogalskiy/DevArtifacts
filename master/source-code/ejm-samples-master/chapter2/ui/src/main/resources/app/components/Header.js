import React from 'react';

export default function(props) {
  return (
    <nav className="navbar navbar-pf-vertical">
      <div className="navbar-header">
        <div className="navbar-toggle">
          <span className="sr-only">Toggle navigation</span>
          <span className="icon-bar"></span>
          <span className="icon-bar"></span>
          <span className="icon-bar"></span>
        </div>
        <a href="/" className="navbar-brand">
          <span className="navbar-brand-name">Address Database</span>
        </a>
      </div>
    </nav>
  );
}
