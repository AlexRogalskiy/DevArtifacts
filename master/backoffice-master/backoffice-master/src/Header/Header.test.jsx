import React from 'react'
import { mount } from 'enzyme'

import Header from './'

it('renders correctly', () => {
  const tree = mount((
    <Header
      isOpen
      title="Header"
      isFixed
      isCookieInfoOpen
      onDrawerOpen={() => {}}
      onClick={() => { }}
    />
  ))

  expect(tree).toMatchSnapshot()
})

it('renders correctly with closed', () => {
  const tree = mount((
    <Header
      title="Header"
      isFixed={false}
      isCookieInfoOpen={false}
      onDrawerOpen={() => { }}
      onClick={() => { }}
    />
  ))

  expect(tree).toMatchSnapshot()
})
