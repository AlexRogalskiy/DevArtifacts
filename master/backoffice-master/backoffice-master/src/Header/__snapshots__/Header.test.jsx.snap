// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="Header-root-1"
>
  <header
    className="MuiPaper-root-17 MuiPaper-elevation4-23 MuiAppBar-root-9 MuiAppBar-positionFixed-10 MuiAppBar-colorPrimary-15 mui-fixed Header-appBar-2 Header-appBarShift-4 Header-appBarWithCookieInfo-3"
  >
    <div
      className="MuiToolbar-root-44 MuiToolbar-regular-46 MuiToolbar-gutters-45"
    >
      <button
        aria-label="open drawer"
        className="MuiButtonBase-root-54 MuiIconButton-root-48 MuiIconButton-colorInherit-49 Header-menuButton-5 Header-hide-6"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiIconButton-label-53"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-57"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
            />
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root-64"
        />
      </button>
      <h2
        className="MuiTypography-root-71 MuiTypography-title-77 MuiTypography-colorInherit-90 Header-flex-7 Header-title-8"
        onClick={[Function]}
      >
        Header
      </h2>
    </div>
  </header>
</div>
`;

exports[`renders correctly with closed 1`] = `
<div
  className="Header-root-1"
>
  <header
    className="MuiPaper-root-17 MuiPaper-elevation4-23 MuiAppBar-root-9 MuiAppBar-positionStatic-13 MuiAppBar-colorPrimary-15 Header-appBar-2"
  >
    <div
      className="MuiToolbar-root-44 MuiToolbar-regular-46 MuiToolbar-gutters-45"
    >
      <button
        aria-label="open drawer"
        className="MuiButtonBase-root-54 MuiIconButton-root-48 MuiIconButton-colorInherit-49 Header-menuButton-5"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiIconButton-label-53"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-57"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
            />
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root-64"
        />
      </button>
      <h2
        className="MuiTypography-root-71 MuiTypography-title-77 MuiTypography-colorInherit-90 Header-flex-7 Header-title-8"
        onClick={[Function]}
      >
        Header
      </h2>
    </div>
  </header>
</div>
`;
